frontend/src/metabase/dashboard/actions/tabs.ts

```diff
@@ -293,9 +293,14 @@ export const tabsReducer = createReducer<DashboardState>(
         }
 
         // 1. Replace temporary with real dashcard ids
-        const prevCards = prevDash.ordered_cards.filter(
-          id => !state.dashcards[id].isRemoved,
-        );
+        const prevCards = prevDash.ordered_cards.filter(id => {
+          const dashcard = state.dashcards[id];
+          const isNotRemoved = !dashcard.isRemoved;
+          const isQuestionCard = dashcard.card_id != null;
```

Comment:
dashcards without card_id are visual cards like heading or text, we shouldn't set `dashcardData` for them